compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -311,9 +316,11 @@ impl<'p, 'tcx> RustcMatchCheckCtxt<'p, 'tcx> {
                 RangeEnd::Included,
             )
         };
+        // Abort on type error.
+        ty.error_reported()?;
```

Comment:
It's a bit unsatisfying that we're aborting very really high up the "type tree" for a possibly really deeply nested error ty. This would cause us to bail even when we have `[[[ty::Error]]]`, etc.

I guess there's nothing necesarily wrong w it, but it does make me a bit happier when we handle things as "tightly" as possible, if that makes sense. For example, in the original ICE, we're trying to match a `ty::Error` against a `Struct` ctor. Why?

Reply:
Hm yeah, in the original ICE, `ty` here was directly `ty::Error`. I'll give it some thought, it's probably ok to only abort in that case

Reply:
Yep that's fine to do